modules/azure/aks/aks.tf

```diff
@@ -230,6 +230,132 @@ resource "azurerm_monitor_diagnostic_setting" "log_storage_account_audit" {
   }
 }
 
+resource "azurerm_monitor_diagnostic_setting" "log_eventhub_audit" {
+  name                           = "eventhub-${var.environment}-${var.location_short}-${var.name}${var.aks_name_suffix}-audit"
+  target_resource_id             = azurerm_kubernetes_cluster.this.id
+  eventhub_name                  = var.log_eventhub_name
+  eventhub_authorization_rule_id = var.log_eventhub_authorization_rule_id
+
+  log {
+    category = "kube-scheduler"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "kube-controller-manager"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "cloud-controller-manager"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "csi-azurefile-controller"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "csi-snapshot-controller"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "csi-azuredisk-controller"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "guard"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+      days    = 0
+    }
+  }
+
+  log {
+    category = "cluster-autoscaler"
+    enabled  = true
```

Comment:
What features do we want to log?

Reply:
That is a really good question

Reply:
Me and @jimgus  have decided that we will start with cluster-autoscaler and we will look in to more options in a future task.